Week3/homework/rabie_todo_app.sql

```diff
@@ -0,0 +1,109 @@
+-- MySQL dump 10.13  Distrib 8.0.18, for macos10.14 (x86_64)
+--
+-- Host: localhost    Database: todo_app
+-- ------------------------------------------------------
+-- Server version	8.0.18
+
+/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
+/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
+/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
+/*!50503 SET NAMES utf8mb4 */;
+/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
+/*!40103 SET TIME_ZONE='+00:00' */;
+/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
+/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
+/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
+/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;
+
+--
+-- Table structure for table `todo_list`
+--
+
+DROP TABLE IF EXISTS `todo_list`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `todo_list` (
+  `todolist_id` int(11) NOT NULL AUTO_INCREMENT,
+  `todolist_name` varchar(50) DEFAULT NULL,
+  `reminder` datetime DEFAULT NULL,
+  `user_id` int(11) DEFAULT NULL,
+  PRIMARY KEY (`todolist_id`),
+  KEY `user_id` (`user_id`),
+  CONSTRAINT `todo_list_ibfk_1` FOREIGN KEY (`user_id`) REFERENCES `users` (`user_id`)
+) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Dumping data for table `todo_list`
+--
+
+LOCK TABLES `todo_list` WRITE;
+/*!40000 ALTER TABLE `todo_list` DISABLE KEYS */;
+INSERT INTO `todo_list` VALUES (1,'Database Homework','2019-10-17 00:00:00',1);
+/*!40000 ALTER TABLE `todo_list` ENABLE KEYS */;
+UNLOCK TABLES;
+
+--
+-- Table structure for table `todos`
+--
+
+DROP TABLE IF EXISTS `todos`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `todos` (
+  `todo_id` int(11) NOT NULL,
+  `todo_name` varchar(255) DEFAULT NULL,
+  `end_date` datetime DEFAULT NULL,
+  `tag` varchar(45) DEFAULT NULL,
+  `done` enum('true','false') DEFAULT NULL,
+  `todolist_id` int(11) DEFAULT NULL,
```

Comment:
I would have personally have chosen a cross table between todo items and the todo lists. This way, items can be re-utilizable among the lists.